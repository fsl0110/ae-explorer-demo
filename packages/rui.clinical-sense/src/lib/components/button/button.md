# Button

### Example

```jsx
<Button type="primary">I'm an Ant-Design Button</Button>
```
