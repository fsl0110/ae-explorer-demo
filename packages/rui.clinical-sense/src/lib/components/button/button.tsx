import React, {FC, ReactNode} from "react";
import {Button as AntButton} from "antd";

export interface Props {
  /**
   * Style of the button
   * @default primary
   */
  type: "default" | "primary" | "danger" | "link";

  /**
   * Children of the button.
   */
  children: ReactNode | ReactNode[];
}

export const Button: FC<Props> = ({type, ...rest}) => (
  <AntButton type={type} {...rest}>
    Button
  </AntButton>
);

export default Button;
