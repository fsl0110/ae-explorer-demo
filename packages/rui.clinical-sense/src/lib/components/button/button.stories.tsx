import React from "react";
import {storiesOf} from "@storybook/react";
import {Button} from "./button";
import button from "./button.md";

storiesOf("Utils|Button", module).add(
  "Button",
  () => <Button type="primary">I'm an Ant-Design Button</Button>,
  {
    notes: {markdown: button},
  },
);
