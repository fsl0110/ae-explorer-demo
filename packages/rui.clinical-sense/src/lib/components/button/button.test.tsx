import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import {Button, Props} from "./button";

const testCases: Array<[string, Props]> = [
  [
    "Button default",
    {
      type: "default",
      children: "some text",
    },
  ],
  [
    "Button primary",
    {
      type: "primary",
      children: "some text",
    },
  ],
  [
    "Button link",
    {
      type: "link",
      children: "some text",
    },
  ],
];

describe("<Heading>", () => {
  testCases.forEach(([item, options]) => {
    it(item, () => {
      expect(toJson(shallow(<Button {...options} />))).toMatchSnapshot();
    });
  });
});
