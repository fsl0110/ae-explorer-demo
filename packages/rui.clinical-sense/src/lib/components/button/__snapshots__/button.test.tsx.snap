// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading> Button default 1`] = `
<Button
  block={false}
  ghost={false}
  htmlType="button"
  loading={false}
  type="default"
>
  Button
</Button>
`;

exports[`<Heading> Button link 1`] = `
<Button
  block={false}
  ghost={false}
  htmlType="button"
  loading={false}
  type="link"
>
  Button
</Button>
`;

exports[`<Heading> Button primary 1`] = `
<Button
  block={false}
  ghost={false}
  htmlType="button"
  loading={false}
  type="primary"
>
  Button
</Button>
`;
